docs/setup.md

```diff
@@ -0,0 +1,111 @@
+
+```bash
+npm install -g pnpm
+```
+
+Now, you will need to have packages installed to start development.
+
+```bash
+pnpm install
+```
+
+> If you encounter with error `EACCES: permission denied, open...'`, please run `chown -R $USER /pathToYourProject/Mask` to solve.
+
+### Start the local development server
+
+For Chromium-based browsers (Chrome, Opera, Edge, etc.), please run `pnpm start`. It's preset of many development commands.
+
+If you need to develop in other environments (for example, Firefox), please run `pnpm run go`. It is an interactive CLI tool to help you learn how to compose the build flags.
+
+### Load the extension into your browser
+
+Mask Network has a huge codebase, and it might take time to let the webpack fully startup. It outcomes the `disk/` folder, which contains the unpacked source files of a development version of the Mask Network extension.
+
+For Chrome,
+
+- Open `chrome://extensions` or `Settings -> Extensions`.
+- Turn on the `Developer mode` on the top right corner.
+- It will present a top toolbar with an action button `Load unpacked` on it. Click it and choose the `dist/` folder to load the unpacked version of the Mask Network extension. You can drag and drop the `dist` folder into this page.
+- If everything goes fine. Then, the Mask Network will guide you to the setup process.
+
+For Firefox, it's quite the same process.
+
+- Open `about:debugging#/runtime/this-firefox`
+- Click the `Load Temporary Add-on…` and select the `dist/` folder to load the unpacked extension.
+- If everything goes fine. The Mask Network will start to guide you to the setup process.
+
+## Debugging
+
+There is no difference between extension development and normal web development. As normal web development, only one front-end page to have, but an extension could have more than one page simultaneously. A background page without a user interface running all the time and an options page just like a normal web page with which users can interact. Moreover, an extension could inject content script into the currently visiting page with the allowance from the users.
+
+### Debug the background page
+
+The background page of the Mask Network extension maintains a bunch of fundamental services for front-end functions. Like Crypto Algorithm, Web3 SDKs, APIs to many third-party data providers, etc. They are stand-by all the time, once to be called for a specific task.
+
+To debug _background service_, click links right after **Inspect views**
+
+![An image displaying Chrome extension manage page](https://user-images.githubusercontent.com/5390719/103509131-5ce0cb00-4e9d-11eb-9aec-b24b9888b863.png)
+
+### Debug the content script
+
+Mask Network only injects content script with the permission approved by the user. For each new SNS website that Mask Network is going to support. It will present a prompt dialog for asking permission dynamically, rather than asking for all mightly permission at the plugin got installed.
```

Comment:
```suggestion
Mask Network only injects content script with permission from the user.

For every new website that Mask Network is going to support, it will show a prompt dialog to ask permission dynamically, rather than asking for all mightly permission at the plugin got installed.
```